resolwe/flow/executors/run.py

```diff
@@ -0,0 +1,322 @@
+""".. Ignore pydocstyle D400.
+
+==============
+Flow Executors
+==============
+
+.. autoclass:: resolwe.flow.executors.BaseFlowExecutor
+    :members:
+
+"""
+from __future__ import absolute_import, division, print_function, unicode_literals
+
+import datetime
+import functools
+import json
+import logging
+import os
+import shutil
+import traceback
+import uuid
+from collections import defaultdict
+
+import redis
+import six
+
+
+class DictObjBridge(dict):
+    """Simple bridge class to unify the access pattern for settings."""
+
+    def __getattr__(self, name):
+        """Intercept setting access."""
+        try:
+            return self.__getitem__(name)
+        except KeyError:
+            raise AttributeError(name)
+
+
+def dict_dot(d, k, val=None, default=None):
+    """Get or set value using a dot-notation key in a multilevel dict.
+
+    Dict-only specialization of resolwe.flow.utils.dict_dot.
+    """
+    if val is None and k == '':
+        return d
+
+    def set_default(dict_or_model, key, default_value):
+        """Set default field value."""
+        return dict_or_model.setdefault(key, default_value)
+
+    def get_item(dict_or_model, key):
+        """Get field value."""
+        return dict_or_model[key]
+
+    def set_item(dict_or_model, key, value):
+        """Set field value."""
+        dict_or_model[key] = value
+
+    if val is None and callable(default):
+        # Get value, default for missing
+        return functools.reduce(lambda a, b: set_default(a, b, default()), k.split('.'), d)
+
+    elif val is None:
+        # Get value, error on missing
+        return functools.reduce(get_item, k.split('.'), d)
+
+    else:
+        # Set value
+        try:
+            k, k_last = k.rsplit('.', 1)
+            set_item(dict_dot(d, k, default=dict), k_last, val)
+        except ValueError:
+            set_item(d, k, val)
+        return val
+
+
+# to ensure uniform access, wrap the json settings into this, because
+# the code below will only access them in a django-like fashion (through attributes, not dict items)
+settings = DictObjBridge(json.load(open('settings.json', 'rt')))  # pylint: disable=invalid-name
+
+now = datetime.datetime.now  # pylint: disable=invalid-name
+
+
+logger = logging.getLogger(__name__)  # pylint: disable=invalid-name
+CWD = os.getcwd()
+
+
+def iterjson(text):
+    """Decode JSON stream."""
+    decoder = json.JSONDecoder()
+    while len(text) > 0:
+        obj, ndx = decoder.raw_decode(text)
+
+        if not isinstance(obj, dict):
+            raise ValueError()
+
+        text = text[ndx:].lstrip('\r\n')
+        yield obj
+
+
+class BaseFlowExecutor(object):
+    """Represents a workflow executor."""
+
+    exported_files_mapper = defaultdict(dict)
+
+    def __init__(self, *args, **kwargs):
+        """Initialize attributes."""
+        self.data_id = None
+        self.process = None
+        # Flag to mark current object as failed.
```

Comment:
Where is this flag?

Reply:
The flag itself was deleted but I missed that comment line.